docs/aws/audit/esmonitoring/rules/es_domain_exposed_remediation.mdx

```diff
@@ -1,6 +1,153 @@
+   ```
+
+By following these steps, you can ensure that your Elasticsearch domain is not publicly exposed and is accessible only from trusted sources.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the Amazon Elasticsearch Service console at https://console.aws.amazon.com/es/.
+
+2. In the navigation pane, choose "Domains". This will display a list of all Elasticsearch domains that are currently available in your AWS account.
+
+3. Select the Elasticsearch domain that you want to examine, then choose the "Modify access policy" option.
+
+4. In the "Modify access policy" section, check the policy document for the "Effect": "Allow" statement and the "Principal": "*" element. If the policy document contains these elements, the selected Elasticsearch domain is publicly accessible, which means it's exposed to the internet.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can run any AWS CLI commands, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided in the AWS CLI User Guide.
+
+2. List all Elasticsearch domains: Use the following AWS CLI command to list all the Elasticsearch domains in your AWS account:
+
+   ```
+   aws es list-domain-names
+   ```
+   This command will return a list of all Elasticsearch domain names.
+
+3. Describe the Elasticsearch domain: For each domain name returned in the previous step, use the following AWS CLI command to get detailed information about the domain:
+
+   ```
+   aws es describe-elasticsearch-domain --domain-name <your-domain-name>
+   ```
+   Replace `<your-domain-name>` with the name of your Elasticsearch domain. This command will return a JSON object that contains detailed information about the domain.
+
+4. Check the access policy: In the JSON object returned in the previous step, look for the `AccessPolicies` field. This field contains the access policy for the Elasticsearch domain. If the access policy allows public access (i.e., access from '0.0.0.0/0' or '::/0'), then the Elasticsearch domain is publicly exposed.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. 
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+```
+
+2. Create a session using your AWS credentials. You can configure your credentials in the AWS credentials file located by default at `~/.aws/credentials`.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>